logistic_order/model/sale_order.py

```diff
@@ -194,6 +200,35 @@ def _get_volume(self):
     def _get_weight(self):
         self.weight = self.product_id.weight * self.product_uom_qty
 
+    def product_id_change_with_wh(self, cr, uid, ids,
+                                  pricelist, product,
+                                  qty=0,
+                                  uom=False,
+                                  qty_uos=0,
+                                  uos=False,
+                                  name='',
+                                  partner_id=False,
+                                  lang=False,
+                                  update_tax=True,
+                                  date_order=False,
+                                  packaging=False,
+                                  fiscal_position=False,
+                                  flag=False,
+                                  warehouse_id=False,
+                                  context=None):
+        res = super(SaleOrderLine, self).product_id_change_with_wh(
+            cr, uid, ids,
+            pricelist, product, qty, uom,
+            qty_uos, uos, name, partner_id,
+            lang, update_tax, date_order,
+            packaging, fiscal_position, flag,
+            warehouse_id,
+            context=context)
+
+        if 'price_unit' in res.get('value', []):
+            res['value']['value_of_goods'] = res['value']['price_unit']
```

Comment:
you should use setdefault('value', []) on line 228 for this to be correct (because otherwise if the key `'value'` is missing in `res`, the call to `get()` will work but `res['value']` will raise. 

Not sure if this is can happen, though?